### ISSUE TITLE
Use kube-router upstream libraries to consume the netpol controller

### ISSUE THREAD

**Issue by manuelbuil** [k3s-io/k3s#5078]
**Is your feature request related to a problem? Please describe.**

We are copying a part of kube-router code to pkg/agent/netpol directory with modifications, from which the biggest one is tje consumption of k3s node config instead of kube-router config.

However, that approach makes it hard to follow new upstream versions. It's possible to use kube-router as a library, so it seems like a better way to do that.

**Describe the solution you'd like**

Import kube-router libraries instead 

**Describe alternatives you've considered**

Nothing

**Additional context**

**Backporting**

- [X] Needs backporting to older releases

**Comment by manuelbuil** [comment]
We can verify we are running the newest code by checking the log "Starting the netpol controller" in journalctl. The old code did not have that

**Comment by ShylajaDevadiga** [comment]
Validated using commit id **5233946c** on master branch

- Journalctl displays the expected log message 

```
$ journalctl -u k3s|grep netpol
Feb 11 05:18:23 ip-172-31-14-23 k3s[1897]: time="2022-02-11T05:18:23Z" level=info msg="Starting the netpol controller"
```

- Validated network policies work as expected.

**Comment by ShylajaDevadiga** [comment]
Validated using k3s version v1.23.4-rc1+k3s1 (43b1cb48)
```
sudo journalctl -u k3s|grep netpol
Feb 22 23:18:02 ip-172-31-0-189 k3s[2240]: time="2022-02-22T23:18:02Z" level=info msg="Starting the netpol controller"
```
